pkg/registry/service/rest.go

```diff
@@ -277,8 +277,8 @@ var _ = rest.Redirector(&REST{})
 
 // ResourceLocation returns a URL to which one can send traffic for the specified service.
 func (rs *REST) ResourceLocation(ctx api.Context, id string) (*url.URL, http.RoundTripper, error) {
-	// Allow ID as "svcname" or "svcname:port".
-	svcName, portStr, valid := util.SplitPort(id)
+	// Allow ID as "svcname", "svcname:port", or "svcname:port:scheme".
```

Comment:
I actually couldn't find any documentation referencing the name:port format... is it outside the repo somewhere?